### ISSUE TITLE
Accessibility: VoiceOver unable to identify or read message and other contents in table cells

### ISSUE THREAD

**Issue by ssawczyn** [overtake/TelegramSwift#380]
I am a blind VoiceOver user.  When I use VoiceOver, I can find two tables within the application, presumably one is the conversation list and the other the contents of the conversation.  Unfortunately, VoiceOver cannot read the cell contents of either table (VoiceOver announces "empty cell" which is inaccurate).  I've seen this behavior before and while I am not a developer, my understanding is that this has to do with the way the tables are coded.  If it would help, I can provide links to apple's accessibility documentation.  I'm hoping this will ultimately be an easy fix as I'd really love to use this client.

**Comment by beqabeqa473** [comment]
+100000

i am also a voice over user and table cells are not accessible.

please fix that behaviour so contents of table cells could be read by voice over.

thanks./

**Comment by MarcoZehe** [comment]
The problem seems to have got worse, right now, VoiceOver is unable to identify anything in the window, even though the initial screen claims to have keyboard focus in the phone number field. I can neither edit it nor set focus to it so that it accepts keyboard focus. I am effectively unable to even add this client to my account right now. Version is 7.4.

**Comment by beqabeqa473** [comment]
Any progress on that topic?

**Comment by ssawczyn** [comment]
I haven’t tried in the last few months, but I also haven’t seen any new activity on this issue which is really unfortunate.  Right now, I don’t have a way to use Telegram from my Mac at all.

On Feb 14, 2023, at 7:01 PM, Beqa Gozalishvili ***@***.***> wrote:



Any progress on that topic?
